inc/zoo/map/RobinHood.h

```diff
@@ -387,62 +336,89 @@ struct RH_Frontend_WithSkarupkeTail {
                     return std::pair{values_.data() + index, true};
                 }
                 // the last element is special because it is a
                 // move-construction, not a move-assignment
-                auto fromIndex = relocations[--relocationsCount];
+                --relocationsCount;
+                auto fromIndex = relocations[relocationsCount];
                 values_[index].build(
                     std::move(values_[fromIndex].value())
                 );
+                md_[swarIndex] =
+                    md_[swarIndex].blitElement(
+                        intraIndex, elementToInsert
+                    );
+                elementToInsert = newElements[relocationsCount];
                 index = fromIndex;
+                swarIndex = index / MD::NSlots;
+                intraIndex = index % MD::NSlots;
                 // do the pair relocations
                 while(relocationsCount--) {
                     fromIndex = relocations[relocationsCount];
                     values_[index].value() =
                         std::move(values_[fromIndex].value());
+                    md_[swarIndex] = md_[swarIndex].blitElement(intraIndex, elementToInsert);
+                    elementToInsert = newElements[relocationsCount];
                     index = fromIndex;
+                    swarIndex = index / MD::NSlots;
+                    intraIndex = index % MD::NSlots;
                 }
                 values_[index].value() = std::pair(k, mv);
+                md_[swarIndex] =
+                    md_[swarIndex].blitElement(intraIndex, elementToInsert);
                 return std::pair{values_.data() + index, true};
             }
+            if(HighestSafePSL < evictedPSL) {
+                throw MaximumProbeSequenceLengthExceeded("Encoding insertion");
+            }
+            
             // evict the "deadline" element:
             // first, insert the element in its place (it "stole")
             // find the place for the evicted: when Robin Hood breaks again.
 
             // for this search, we need to make a search needle with only
             // the PSL being evicted.
 
-            // The needle stole the entry: replace it with the needle
-            assignMetadataElement(
-                deadline,
-                elementToInsert.shiftLanesLeft(intraIndex),
-                mdp
-            );
-            // now the insertion will be for the old metadata entry
-            elementToInsert = MD(mdBackup.hashes().at(intraIndex));
             // "push" the index of the element that will be evicted
-            relocations[relocationsCount++] = index;
+            relocations[relocationsCount] = index;
+            // we have a place for the element being inserted, at this index
+            newElements[relocationsCount++] = elementToInsert;
+            if(MaxRelocations <= relocationsCount) {
+                throw RelocationStackExhausted("");
```

Comment:
We need a solution at some point in the future for relocation stack exhaustion (of some sort).  It'll constrain maximum load factor for tables, esp large ones.

Reply:
For demonstration purposes we can use however large a factor let's us do the work.
This is something that depends in an unresolved question, of journal/no journal; strong/not strong exception guarantees.